crates/oxc_linter/src/rules/typescript/no_namespace.rs

```diff
@@ -157,7 +157,9 @@ impl Rule for NoNamespace {
             return;
         }
 
-        if self.allow_declarations && is_declaration(node, ctx) {
+        if self.allow_declarations
+            && (declaration.declare || is_any_ancestor_declaration(node, ctx))
+        {
```

Comment:
in future, it would be great if we could leave changes like this out.

The only reason is I just can't see why we're making this change (before `declaration.declare` didn't exist, do we have a tests for this, ect)

Reply:
see other comment